src/solver/navier_stokes/bc.jl

```diff
@@ -43,14 +45,14 @@ function (obj::ExactChannelBC)(
               bndryflux::AbstractArray{Tres, 1}) where {Tmsh, Tsol, Tres}
 
   sigma = 0.01
-  gamma = params.gamma
-  gamma_1 = params.gamma - 1
-  aoa = params.aoa
+  gamma = params.euler_params.gamma
+  gamma_1 = params.euler_params.gamma - 1
+  aoa = params.euler_params.aoa
   beta = params.sideslip_angle
   rhoInf = 1.0
-  uInf = params.Ma * cos(beta) * cos(aoa)
-  vInf = params.Ma * sin(beta) * -1
-  wInf = params.Ma * cos(beta) * sin(aoa)
+  uInf = params.euler_params.Ma * cos(beta) * cos(aoa)
```

Comment:
Please check if the angle definitions here are the same as those in EulerMod.

Reply:
There is not a corresponding function in the Euler module.  After looking over the boundary conditions in the NS module, I am now more confused than before.  There are some boundary functors in `bc.jl` and some `AbstractBoundaryValueType` in `viscous_func.jl`, but I'm not sure what the differnece is.  Are the ones in `bc.jl` for the inviscid part and the the ones in `viscous_func.jl` for the viscous part?

Reply:
`AbstractBoundaryValueType` and its derived types should be put into `bc.jl`.

`AbstractBoundaryValueType` is more like computing `qR` while funcitons in `bc.jl` are used to compute fluxes directly.

Reply:
Ok, I have moved the `AbstractBoundaryValueType` stuff into `bc.jl`.  I also moved the boundary condition flux calculation from `flux.jl` to `bc.jl`, so all BC stuff is in one place.

I also added some comments to `bc.jl` saying that I think `FreeStreamBC` is the only one that works right now (I do not see the inviscid part of other boundary conditions in the Euler module, so if anyone tries to use them they will get an error message).  Basically this is a warning that whoever works on the viscous code next should take a careful look at the boundary conditions.